include/swift/AST/DiagnosticEngine.h

```diff
@@ -944,11 +953,15 @@ namespace swift {
     void setSuppressRemarks(bool val) { suppressRemarks = val; }
     bool getSuppressRemarks() const { return suppressRemarks; }
 
-    /// Whether a warning should be upgraded to an error or not
-    void setWarningAsErrorForDiagID(DiagID id, bool value) {
+    /// Sets whether warnings belonging to the diagnostic group identified by
+    /// `id` should be escalated to errors.
+    void setWarningsAsErrorsForDiagGroupID(DiagGroupID id, bool value) {
```

Comment:
It took me a moment to understand why it's a valid change for the case when we apply `WarningAsErrorRule` with `TargetAll`. But it seems it's ok as all the uncategorized warnings have the `no_group` group which will be stored in this BitVector like the others.
I'd leave a comment about it though.

Reply:
Sure. Will add a test too.